test/TestAssets/BuildWebHostPatternTestSite/Program.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Hosting;
+
+namespace BuildWebHostPatternTestSite
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+        }
+
+        public static IWebHost BuildWebHost(string[] args) => null;
```

Comment:
null? Your tests should execute the resolved factory to make sure it returns the expected result.

Reply:
I don't have a server so I can't build a WebHost. The logic is only about finding the method, what it does when called is the responsibility of the CreateWebHostBuilder method.

Reply:
You're creating a factory, but you don't test that your factory works correctly. Returning a fake IWebHost implementation would be adequate.